src/openforms/forms/tasks.py

```diff
@@ -87,6 +105,33 @@ def recouple_submission_variables_to_form_variables(form_id: int) -> None:
         logger.info("Form variables were updated while this task was runnning.")
 
 
+@app.task(ignore_result=True)
+def recouple_reusable_definition_variables_to_form_variables(form_id: int) -> None:
+    """
+    When the FormVariable bulk create/update endpoint is called, all existing FormVariable related to the form are
+    deleted and new are created. If there are any form definitions which are reusable, we want to update all the forms
+    which are also using these FormDefinitions (concerning the FormVariables). This task updates the FormVariables
+    of each related form in the database, by deleteing the old ones and recreating them (the most safe way).
+    """
+    from .models import FormDefinition, FormVariable  # due to circular import
+
+    form = Form.objects.get(id=form_id)
+    reusable_form_definitions = FormDefinition.objects.filter(
+        formstep__form=form, is_reusable=True
+    ).distinct()
+
+    for fd in reusable_form_definitions:
+        form_ids_to_update = fd.formstep_set.exclude(form=form).values_list(
+            "form", flat=True
+        )
+
+        for form_id in form_ids_to_update:
+            form_to_update = Form.objects.get(id=form_id)
+            form_to_update.formvariable_set.all().delete()
```

Comment:
this is also destroying the user defined variables, which is *not* okay. Please add a test case of a form with user defined variables too and assert that those are not broken.

Reply:
Yes, this was done by accident. At the beginning I was filtering out only `FormVariableSources.component` sources but yes, I will add a test for this as well.